Server/src/controllers/authenticationController.ts

```diff
@@ -86,16 +96,15 @@ export class AuthenticationController {
   }
 
   private validateUpdatePasswordRequest(request: Request): boolean {
-    return !request.body.password && !request.body.passwordConfirmation && request.body.password === request.body.passwordConfirmation;
+    return !passwordSchema.isValidSync(request.body.password) && !request.body.passwordConfirmation && request.body.password === request.body.passwordConfirmation;
```

Comment:
You should switch the verifications here. Since each statement is sequentially evaluated

So doing request.body.password && request.body.password confirmation then the checking the schema

You want to check first that the fields are input then verify if they match the schema

Reply:
should it be something like :
!request.body.password && !request.body.passwordConfirmation && request.body.password === request.body.passwordConfirmation && !passwordSchema.isValidSync(request.body.password);
?